src/WaveFunction/WaveFunction.cpp

```diff
@@ -62,4 +63,188 @@ double WaveFunction::calculateShannonEntropy() const {
 }
 
 
+/**
+ *  Transform the underlying ONV basis of the wave function (only for FCI [ProductFockSpace]) as shown in Helgaker 11.9
+ *
+ *  @param T    the transformation matrix between the old and the new orbital basis
+ */
+void WaveFunction::basisTransform(const SquareMatrix<double>& T) {
+
+    if (fock_space->get_type() != FockSpaceType::ProductFockSpace) {
+        throw std::invalid_argument("WaveFunction::basisTransform(SquareMatrix<double>): This is not an FCI wave function");
+    }
+
+    auto K = fock_space->get_K();
+
+    if (K != T.get_dim()) {
+        throw std::invalid_argument("WaveFunction::basisTransform(SquareMatrix<double>): number of orbitals does not match the dimension of the transformation matrix T");
+    }
+
+    // Retrieve LU decomposition for T
+    const auto& LU = T.NoPivotLUDecomposition();
+
+    // Retrieve L
+    SquareMatrix<double> L = SquareMatrix<double>::Identity(K, K);
+    L.triangularView<Eigen::StrictlyLower>() = LU[0];
+
+    // Retrive U
+    SquareMatrix<double> U = SquareMatrix<double>(LU[1].triangularView<Eigen::Upper>());
+
+
+    SquareMatrix<double> U_in = U.inverse();
```

Comment:
```suggestion
    SquareMatrix<double> U_inv = U.inverse();
```

I'd rename to `inv` because `in` has other meanings, like 'input' or something